token-metadata/program/src/state.rs

```diff
@@ -102,19 +114,19 @@ pub trait TokenMetadataAccount {
         Ok(())
     }
 
-    fn safe_deserialize<T: BorshDeserialize>(mut data: &[u8]) -> Result<T, BorshError> {
-        if !is_correct_account_type(data, Self::key(), Self::size()) {
+    fn safe_deserialize(mut data: &[u8]) -> Result<Self, BorshError> {
+        if !Self::is_correct_account_type(data, Self::key(), Self::size()) {
             return Err(BorshError::new(ErrorKind::Other, "DataTypeMismatch"));
         }
 
-        let result: T = T::deserialize(&mut data)?;
+        let result = Self::deserialize(&mut data)?;
 
         Ok(result)
     }
 
-    fn from_account_info<T: BorshDeserialize>(a: &AccountInfo) -> Result<T, ProgramError>
+    fn from_account_info(a: &AccountInfo) -> Result<Self, ProgramError>
 where {
-        let ua: T = Self::safe_deserialize(&a.data.borrow_mut())
+        let ua: Self = Self::safe_deserialize(&a.data.borrow_mut())
```

Comment:
Nit: was this `ua:  Self` still needed now that safe deserialize specifically returns `Result<Self, BorshError>`?